### PR TITLE
Ensure `ConfigurationCacheClassLoaderScopeRegisteryListener` is always disposed of

### PR DESCRIPTION
**Pull request by bamboo**
As soon as possible.

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by bamboo** [comment]
@bot-gradle test RFN

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheServices.kt [hunk 1]
```diff
@@ -73,17 +74,26 @@ class ConfigurationCacheServices : AbstractPluginServiceRegistry() {
 }
 
 
+class ConfigurationCacheBuildEnablement(
```
Comment (eskatos) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheServices.kt:77]: :o: move to its own file?

**Review by eskatos — APPROVED** [review]
LGTM!
:+1:  for the refinements done along the way

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheServices.kt (hunk 1 above)
Comment (bamboo) [reply subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheServices.kt]: Makes sense, thanks!

**Comment by bamboo** [comment]
@bot-gradle test QF

**Pull request merged**